app/roles/date_filtered_dataset.rb

```diff
@@ -12,13 +12,13 @@ class DateFilteredDataset < RolePlaying::Role
   # @return [Sequel::Dataset] a filtered dataset.
   def filter_by_date(from: nil, to: nil, with_original_date: false)
 
-    condition = ->(col, op) do
-      "#{col} #{op} :date".tap do |sql|
-        sql << " OR original_#{col} #{op} :date" if with_original_date
-      end
+    condition = ->(col, op, date) do
+      Sequel.lit("#{col} #{op} :date".tap { |sql|
+          sql << " OR original_#{col} #{op} :date" if with_original_date
+        }, date: date)
```

Comment:
This has become quite unreadable and it won't work with frozen literals. I suggest something like:

```ruby
condition = ->(col, op, date) do
  sql = "#{col} #{op} :date"
  sql += " OR original_#{col} #{op} :date" if with_original_date
  Sequel.lit(sql, date: date)
end